File: fastapi-funds/app/services/notifications_service.py
from fastapi import HTTPException
from twilio.rest import Client
from app.core.config import settings

class NotificationsService:
    @staticmethod
    async def send_sms(to: str, body: str) -> None:
        try:
            client = Client(settings.twilio_account_sid, settings.twilio_auth_token)
            message = client.messages.create(
                body=body,
                from_=settings.twilio_phone_number,
                to=to
            )
        except Exception as error:
            print(f'Error sending SMS: {error}')
            raise HTTPException(status_code=500, detail="Failed to send SMS")

File: funds-backend/src/modules/notifications/notifications.service.ts
import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import * as twilio from 'twilio';

@Injectable()
export class NotificationsService {
  private twilioClient: twilio.Twilio;

  constructor(private configService: ConfigService) {
    const accountSid = this.configService.get<string>('TWILIO_ACCOUNT_SID');
    const authToken = this.configService.get<string>('TWILIO_AUTH_TOKEN');
    this.twilioClient = twilio(accountSid, authToken);
  }

  async sendSMS(to: string, body: string): Promise<void> {
    try {
      await this.twilioClient.messages.create({
        body: body,
        from: this.configService.get<string>('TWILIO_PHONE_NUMBER'),
        to: to,
      });
    } catch (error) {
      console.error('Error sending SMS:', error);
      throw new Error('Failed to send SMS');
    }
  }

  async sendEmail(to: string, body: string) {
    
  }
}